### ISSUE TITLE
Question: Adding a modem for unsupported carriers to get Volte

### ISSUE THREAD

**Issue by Turbolqk** [whatawurst/android_device_sony_yoshino-common#98]
Hey.
From what I read the CPU should be the same when using .mbn files from other devices. Some phones with Snapdragon 835 support Volte for Yettel Bulgaria (formerly Telenor). Do you think it would work to add it in Modem switcher?
I can find the mbn if it would.

**Comment by derfelot** [comment]
As far as I know, the Sony implementation for mbn's (which requires their own "modemswitcher" binary to apply mbn's) requires them to be digitally signed by Sony. They are also not interchangeable between devices (I think I once tested a poplar mbn on lilac), so I don't think this would work.

What you could try is to use a more generic mbn that we do have For example, people have used ir51_ir92_ims_gte for carriers that aren't supported in stock and I added IMS support for them (e.g. see here: https://github.com/derfelot/android_device_sony_maple_dsds/commit/deb98e1120163cd60c18925a36c5c84b9f8a57a7). You can manually apply them using the "Xperia Parts -> Modem Switcher Activity" menu. This will not "fully" enable IMS, but you should be able to check in "Phone Information" (access by dialing *#*#4636#*#*) if IMS is registered.

If his works, I can make a test build with other changes applied to fully enable IMS support for your carrier.

Cheers

**Comment by Turbolqk** [comment]
Hey, thanks for the detailed answer.
This .mbn you pointed out doesn't work for me, but users of other devices report that using Telenor Denmark/Norway/Sweden works for Telenor Bg as well.
For some reason though, these modems present in our device doesn't register ims. It's probably specific to these particular Sony modems.
I think they are the equivalent of the china example you gave. What would you need me to do to add ims?

**Comment by derfelot** [comment]
Which device is this?

**Comment by Turbolqk** [comment]
Nevermind that, I got lucky! amss_fsg_lilac_vodafone_in_ims_tar.mbn reports ims:registered!

**Comment by derfelot** [comment]
ok, but we still need to enable this in overlays, otherwise you probably wont have all options available in settings (e.g. turning volte on/off, turning vowifi on/off etc). 

So I would need to know your carrier details like mcc, mnc and (if available) gid and/or apn (so we can make selecting the modem as specific as possible). Also, it would be good if you could test if vowifi actually works, if not it would better do disable it in overlays.

**Comment by derfelot** [comment]
these settings i mean:

![Screenshot_20230517-172853_Settings.png](https://github.com/whatawurst/android_device_sony_yoshino-common/assets/5124143/b0925d1a-782b-4284-a224-6b588d1c6405)

**Comment by Turbolqk** [comment]
Alright, I'll check them out and write back.

**Comment by Turbolqk** [comment]
Things seem to be more complicated, I'll explain in detail.

Using the modem "amss_fsg_lilac_vodafone_in_ims_tar.mbn" on it's own doesn't report IMS: Registered. I got really lucky though, and had a Magisk module - https://forum.xda-developers.com/t/module-v17-x-volte-vowifi-enabler-2018-09-20.3649613/. Basically this edits build.prop adding/changing some lines (pointed out in the thread). With this module and modem combination, this is how things look:
![photo_2023-05-17_19-04-09](https://github.com/whatawurst/android_device_sony_yoshino-common/assets/17542524/1391bad3-bff1-48c8-97b3-b317b84cb10f)

Due to the module, I have the options you mention, however I can't use VoLTE/ViLTE/VoWIFI in reality. So it doesn't do anything in reality, except it registers IMS for some reason??
![photo_2023-05-17_19-07-35](https://github.com/whatawurst/android_device_sony_yoshino-common/assets/17542524/013a9891-b728-41cb-9d86-1979367d51a4)

I hope this doesn't complicate things.

My carrier supports VoLTE and ViLTE, but not VoWIFI. The carrier details you asked are:
MCC: 205
MNC: 05
I couldn't find GID

APN is as follows:
Name: Yettel Internet
APN: internet
Username: Yettel

In the rom there are currently two APN's from the same network provider, only with outdated names (since they change ownership):
 - Globul internet - the default APN when loading the sim and it works without issue;
 - Telenor internet - half empty junk apn;

Edit: Other modems, like telenor denmark, don't register ims with the module.

**Comment by derfelot** [comment]
yeah, so those props that are set by the module are basically overrides. if there is proper ims registration, they arent neded.

ill see what i can do. ill set things up in a way that of ims registers properly, you can see these options.

**Comment by derfelot** [comment]
~~btw, are you sure your carrier generally supports ims?~~ nevermind, missed part of your previous message

**Comment by Turbolqk** [comment]
Alright, thanks. Why doesn't IMS register without the Magisk module? Modem only doesn't work. Any ideas?

Edit: Nevermind, It's working. For some reason it didn't reguster without the module before. Well now it's certain: ims is registered with this modem.
